### PR TITLE
Implementation of CodableResultClosure with QueryParameters

### PR DESCRIPTION
**Pull request by saiHemak**
In this PR I am implementing the support for CodableResultClosure with QueryParameters

## Description
At the moment the router handler forces the users to use the CodableArrayResultClosure type only. Compilation fails if the application is expecting a Simple Codable type with QueryParameters.

## Motivation and Context
Application needs a Simple JSON object instead of Array of JSON objects.As the current implementation does not allow a SimpleCodable type to be returned , I have implemented the same.

## How Has This Been Tested?
Ran Kitura test-suite with my changes 

## Checklist:

- [ ] If applicable, I have updated the documentation accordingly.
- [ x] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Tests/KituraTests/TestCodableRouter.swift [hunk 1]
```diff
@@ -39,7 +39,8 @@ class TestCodableRouter: KituraTest {
             ("testRouteParameters", testRouteParameters),
             ("testCodableRoutesWithBodyParsingFail", testCodableRoutesWithBodyParsingFail),
             ("testCodableGetQueryParameters", testCodableGetQueryParameters),
```
Comment (ianpartridge) [inline Tests/KituraTests/TestCodableRouter.swift]: Please rename this test to "testCodableGetSingleQueryParameters", then follow it with your new "testCodableGetArrayQueryParameters" test, then finish the array with the existing "testCodableDeleteQueryParameters" test.

That way the test naming is consistent with the non-queryparams cases.

**Comment by saiHemak** [comment]
@ianpartridge   I have renamed the new test case to ```testCodableGetSingleQueryParameters```  and ```testCodableGetQueryParameters```` to ```testCodableGetArrayQueryParameters``` . Please review

**Comment by djones6** [comment]
@saiHemak is this ready to merge?

**Comment by saiHemak** [comment]
@djones6 Changes were approved.Can you please merge

**Pull request merged**